WR31/doormon.py

```diff
@@ -0,0 +1,159 @@
+"""
+Monitor the WR31 door enclosure
+"""
+############################################################################
+#                                                                          #
+# Copyright (c)2015, Digi International (Digi). All Rights Reserved.       #
+#                                                                          #
+# Permission to use, copy, modify, and distribute this software and its    #
+# documentation, without fee and without a signed licensing agreement, is  #
+# hereby granted, provided that the software is used on Digi products only #
+# and that the software contain this copyright notice,  and the following  #
+# two paragraphs appear in all copies, modifications, and distributions as #
+# well. Contact Product Management, Digi International, Inc., 11001 Bren   #
+# Road East, Minnetonka, MN, +1 952-912-3444, for commercial licensing     #
+# opportunities for non-Digi products.                                     #
+#                                                                          #
+# DIGI SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED   #
+# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A          #
+# PARTICULAR PURPOSE. THE SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, #
+# PROVIDED HEREUNDER IS PROVIDED "AS IS" AND WITHOUT WARRANTY OF ANY KIND. #
+# DIGI HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,         #
+# ENHANCEMENTS, OR MODIFICATIONS.                                          #
+#                                                                          #
+# IN NO EVENT SHALL DIGI BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT,      #
+# SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS,   #
+# ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF   #
+# DIGI HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.                #
+#                                                                          #
+############################################################################
+
+import time
+import sys
+import sarcli
+import idigidata
+
+
+def millisecond_timestamp():
+    """
+    Return a timestamp, in milliseconds
+    :return ms_timestamp: int, Timestamp in milliseconds
+    """
+    ms_timestamp = int(time.time() * 1000)
+    return ms_timestamp
+
+
+def cli_command(cmd):
+    """
+    Send a command to the SarOS CLI and receive the response
+    :param cmd: str, Command to run
+    :return response: str, Response to cmd
+    """
+    cli = sarcli.open()
+    cli.write(cmd)
+    response = cli.read()
+    cli.close()

```

Comment:
Thanks Paul, I'll make the changes.

On Mon, Jan 25, 2016 at 3:18 PM, Paul Osborne <notifications@github.com>
wrote:

> In WR31/doormon.py
> <https://github.com/digidotcom/transport_examples/pull/2#discussion_r50757541>
> :
>
> > +    :return ms_timestamp: int, Timestamp in milliseconds
> > +    """
> > +    ms_timestamp = int(time.time() * 1000)
> > +    return ms_timestamp
> > +
> > +
> > +def cli_command(cmd):
> > +    """
> > +    Send a command to the SarOS CLI and receive the response
> > +    :param cmd: str, Command to run
> > +    :return response: str, Response to cmd
> > +    """
> > +    cli = sarcli.open()
> > +    cli.write(cmd)
> > +    response = cli.read()
> > +    cli.close()
>
> This should either be a context manager or, if this device is Python 2.4,
> a try/finally block. E.g.
>
> cli = sarcli.open()try:
>     cli.write(cmd)
>     response = cli.read()finally:
>     cli.close()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/digidotcom/transport_examples/pull/2/files#r50757541>.
>